### DISCUSSION TITLE
SpaCy predictions for specific labels instead of predictions for all labels

### DISCUSSION THREAD

**Discussion by ghost** [explosion/spaCy#11761] [Help: Model Advice] [answered]
Hi.
I am working to train my spaCy model and I want my NER model to make predictions of entities for specific labels only. However, currently, it is returning predictions for all available labels. Please tell me how (if possible) I can do so.

**Comment by polm** [comment, marked as answer]
You can just remove annotations of other entity types, like so:

```
ents = []
for ent in doc.ents:
    if ent.label_ == "PER":
        ents.append(ent)
doc.ents = ents
```

You can't turn off the other entity types - after being trained together the label predictions rely on each other and aren't separable.

Out of curiosity, why do you want to do this?

**Reply by ghost** [reply]
I wish for my NER model to take less time (than it is taking now) in making its predictions. I thought maybe focusing my predictions on specific labels might help.
But your answer seems to make sense since the context of each word is what really matters in predicting its label. So each word would still need to be processed, regardless of whether we want its predictions or not.

Thank you for your kind and quick reply!
